### ISSUE TITLE
ValueError: decoding with 'hex' codec failed (ValueError: string argument should contain only ASCII characters)

### ISSUE THREAD

**Issue by spuddybuddy** [noDRM/DeDRM_tools#690]
### Question / bug report

Trying to use the plugin, throws an exeption.  

### Which version of Calibre are you running?

7.22

### Which version of the DeDRM plugin are you running?

v10.0.3

### If applicable, which version of the Kindle software are you running?

n/a

### Log output

```log
 ./calibre-debug -g
calibre 7.22  embedded-python: True
macOS-15.1-arm64-arm-64bit Darwin ('64bit', '')
('Darwin', '24.1.0', 'Darwin Kernel Version 24.1.0: Thu Oct 10 21:02:45 PDT 2024; root:xnu-11215.41.3~2/RELEASE_ARM64_T8112')
Python 3.11.5
OSX: ('15.1', ('', '', ''), 'arm64')
Interface language: None
EXE path: /Applications/calibre.app/Contents/MacOS/calibre-debug
Successfully initialized third party plugins: DeACSM (0, 0, 16) && DeDRM (10, 0, 3)
QPA platform: cocoa
devicePixelRatio: 2.0
logicalDpi: 72.0 x 72.0
physicalDpi: 128.5000019298764 x 128.5000019298764
ApplicationPaletteChange event ignored
Using calibre Qt style: True
[0.00] Starting up...
[0.01] Showing splash screen...
[1.19] splash screen shown
[1.19] Initializing db...
[3.73] db initialized
[3.73] Constructing main UI...
[4.01] GUI main window shown
[4.03] main UI initialized...
[4.03] Hiding splash screen
[4.04] splash screen hidden
[4.04] Started up in 4.04 seconds with 166 books
2024-12-05 16:18:07.654 calibre-debug[34857:1573277] +[IMKClient subclass]: chose IMKClient_Modern
2024-12-05 16:18:14.848 calibre-debug[34857:1573277] The class 'NSOpenPanel' overrides the method identifier.  This method is implemented by class 'NSWindow'
DeACSM v0.0.16: Trying to parse file Sapiens_9780062316103_1553018.acsm
DeACSM v0.0.16: Try to fulfill ...
Notifying server ...
Notifying server https://acsepub.contentreserve.com/fulfillment/FulfillmentNotification
Fulfillment notification successful.
Notifying optional server https://ofs.contentreserve.com/ACSNotificationService.svc/Notify
Weird Fulfillment Notification response: 
DeACSM v0.0.16: Downloading book ...
DeACSM v0.0.16: Loading book from https://acs.cdn.overdrive.com/ACSStore1/0293-1/59F/E2F/96/{59FE2F96-69BD-4E07-9EFB-61A8C64628E3}Fmt410.epub
Download took 756 ms (HTTP 200)
That's a ZIP file -> EPUB
DeACSM v0.0.16: File successfully fulfilled ...
DeACSM v0.0.16: Executing plugin DeDRM ...
DeDRM v10.0.3: Trying to decrypt lyrsalhz.epub
DeDRM v10.0.3: Verifying zip archive integrity
DeDRM v10.0.3: lyrsalhz.epub is a secure Adobe Adept ePub for UUID 79eefc0d-1432-41b7-bab8-aac7182933a8
DeACSM v0.0.16: Running file type plugin failed with traceback:
Traceback (most recent call last):
  File "encodings/hex_codec.py", line 19, in hex_decode
ValueError: string argument should contain only ASCII characters

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "calibre_plugins.deacsm.__init__", line 486, in run
    plugin_ret = plugin.run(rpl)
                 ^^^^^^^^^^^^^^^
  File "calibre_plugins.dedrm.__init__", line 1067, in run
    decrypted_ebook = self.ePubDecrypt(path_to_ebook)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "calibre_plugins.dedrm.__init__", line 533, in ePubDecrypt
    userkey = codecs.decode(userkeyhex, 'hex')
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: decoding with 'hex' codec failed (ValueError: string argument should contain only ASCII characters)
DeACSM v0.0.16: Plugin returned nothing - skipping
Added Sapiens to db in: 0.1
Added 1 books in 2.9 seconds

```

**Comment by spuddybuddy** [comment]
I don't know what format you are expecting for the value you're decoding here.  In my dedrm.json file, it's a binary string (not  ASCII).   I'm pasting my dedrm.json file below (with only the first few bytes of the key contents).

```
{
  "adeptkeys": {
    "79eefc0d-1432-41b7-bab8-aac7182933a8_AdobeID_lobbyseven@gmail.com": "0�\u0002]\u0002\u0001\u0000<snip>"
  },
  "adobe_pdf_passphrases": [],
  "adobewineprefix": "",
  "androidkeys": {},
  "bandnkeys": {},
  "configured": true,
  "deobfuscate_fonts": true,
  "ereaderkeys": {},
  "kindlekeys": {},
  "kindlewineprefix": "",
  "lcp_passphrases": [],
  "pids": [],
  "remove_watermarks": false,
  "serials": []
}
```

**Comment by ElleKayEm** [comment]
Update to DeDRM 10.0.9: https://github.com/noDRM/DeDRM_tools/releases/tag/v10.0.9

**Comment by spuddybuddy** [comment]
That fixed it, thank you for the prompt response 😄 

Maybe worth doing a dot release with that fix cherry-picked  🍒  if 10.0.9 is going to be beta for a while ?

**Comment by ElleKayEm** [comment]
See noDRM's post about leaving 10.0.9 as beta: https://www.mobileread.com/forums/showpost.php?p=4406053&postcount=1857
